feat(self-hosted): convert experimental vars to config options (#29137)
renovatebot/renovate: docs/usage/self-hosted-configuration.md

Read the [Docker Docs, configure runtime resource constraints](https://docs.docker.com/config/containers/resource_constraints/) to learn more.

> ## dockerMaxPages

Reviewer: There's a typo in your PR title.

> If set to an integer, Renovate will use this as max page number for docker tags lookup on docker registries, instead of the default 20 pages.

Reviewer: ```suggestion
By default, Renovate will fetch a maximum of 20 pages when looking up Docker tags on Docker registries.

If set to an integer, Renovate will use this value as the maximum page number.
```

> This is useful for registries which ignores the `n` parameter in the query string and only return 50 tags per page.

Reviewer: ```suggestion
Setting a different limit is useful for registries that ignore the `n` parameter in Renovate's query string and thus only return 50 tags per page.
```

By setting `githubTokenWarn` to `false`, Renovate suppresses these warnings on Pull Requests, etc.
Disabling the warning is helpful for self-hosted environments that can't access the `github.com` domain, because the warning is useless in these environments.

## gitlabAutoMergeableCheckAttempts

> If set to an positive integer, Renovate will use this as the number of attempts to check if a merge request on GitLab is mergeable before trying to automerge.

Reviewer: ```suggestion
If set to a positive integer, Renovate tries this many times to check if a merge request on GitLab is mergeable, before trying to automerge.
```

> The formula for the delay between attempts is `gitlabAutoMergeableCheckAttempts * attempt * attempt` milliseconds.

Reviewer: ```suggestion
Renovate calculates the delay between attempts with this formula: `gitlabAutoMergeableCheckAttempts * attempt * attempt` milliseconds.
```

Default value: `5` (attempts results in max. 13.75 seconds timeout).

## gitlabBranchStatusDelay

Adjust default time (in milliseconds) given to GitLab to create pipelines for a commit pushed by Renovate.

Can be useful for slow-running, self-hosted GitLab instances that don't react fast enough for the default delay to help.

Default value: `1000` (milliseconds).

## gitlabMergeRequestDelay

> If set, Renovate will use this as a delay to proceed with an automerge.

Reviewer: ```suggestion
If set, Renovate will wait this extra time before an automerge.
```

Maybe something like this is easier to read.

Author: I think delay is more commonly used.

## platform

## platformVersion

> Specify this string for Renovate to skip API checks and provide GitLab/Bitbucket server version directly.

Reviewer: ```suggestion
Set a string here to let Renovate skip some API checks.
Renovate will use this value as the GitLab or Bitbucket server version.
```

Maybe we should explain which API check(s) Renovate will skip?

I think the key point is that renovate skips a "fetch version" API call?

## platform

> ## platformVersion

Reviewer 2: Should be inside `platformOptions`. It's only used inside `initPlatform`

## platform

## platformVersion

Set a string here to let Renovate skip some API calls to fetch the platform version.
Renovate will use this value as the GitLab or Bitbucket server version.

Particularly useful with GitLab's `CI_JOB_TOKEN` to authenticate Renovate or to reduce API calls for Bitbucket.

> Read [platform details](modules/platform/gitlab/index.md) to learn why we need the server version on GitLab.

Reviewer: Is this the correct format for the link? I usually mess this up, but often the links have `./` or `../` in front of them.

Author: It works fine in the [live docs](https://docs.renovatebot.com/self-hosted-experimental/#renovate_x_platform_version)
